cmd/generate_example.go

```diff
@@ -10,7 +10,12 @@ import (
 )
 
 func generateExampleRun(cmd *cobra.Command, args []string) error {
-	examplePath := args[0]
+	var examplePath string
+	if len(args) == 0 {
```

Comment:
Haven't looked closely, but this seems to suggest that cobra doesn't have a feature for defaulting values? @ArmanMazdaee

Reply:
I took a look at cobra's documentation, and it doesn't look like the have any default value support. If @ArmanMazdaee knows more though I'd be glad to change it to an implementation that takes advantage of cobra's features!

Reply:
As I know cobra does not provide any built-in method for providing default values of a positional argument and I think this a design desition. Normally positional argument does not have a default value. Either they are required and should be proved or providing them would change how the command would work. I think now that we have decided to provide the default value for arguments it is time to switch from positional arguments to using flags. This way a user could benefit from a default value without any ambiguity.

Reply:
I think arman's reasoning is valid here. The code would become harder to read and usually flags are used to provide default values. Would love @erbesharat 's take as well here.

Flags make defaulting easier as each flag has a default value or at least can have one.

Reply:
That sounds good to me as well, particularly to avoid ambiguity with commands. Will the flags be something like `--default` ? Also, will we track these changes in this PR/issue or open up a new one?

Reply:
The flags would be something along the lines of:
```
generateExampleRun.Flags().StringVarP(&Output, "output", "o", "resume_example.json", "Example resume output file")
$renderFunction.Flags().StringVarP(&Resume, "resume", "d", "resume.json", "Resume data file")
$renderFunction.Flags().StringVarP(&Output, "output", "o", "resume.html", "Resume data file")
```
-d/--resume for data file (thoughts?)

--output --resume for render example: `resumic render -d resume.json -o resume.html`
--output for generation of examples, uischema, jsonschema example: `resumic generate example -o resume_example.json`
--resume for validate example: `resumic validate -d resume.json`

Reply:
I like the direction this is going! Will we still want to have default values for inputs and outputs, or indicate each input and output using flags each time?

Reply:
From a rough view of the docs from cobra the third string such as "resume.json" is the default. So if the flag isn't provided this value is used. Happy to be corrected on this so.

Reply:
I've realized I'm a bit confused: could you give an example of what running the commands with defaults would look like?

Reply:
`resumic render` would be the equivalent of `resumic render -i resume.json -o resume.html` as an example.

Reply:
Gotcha! So the flags would be input when the user wants to use filenames that are different than the default. Thanks for the clarification!

Reply:
As far as the flags go, I'm in favor of doing the `-r` / `--resume` flag for the resume input, and the `-o` / `--output` for the output. What do y'all think?

Reply:
@ArmanMazdaee what's your take?

generally I'm happy with -r/--resume and -o/--output as well.

Reply:
Hey y'all, just wanted to check back in on this one. @ArmanMazdaee any opinion on the flag?

Reply:
I'm going to go ahead and add this functionality using these flags; if we change our mind about the particular flag letter, it should be an easy fix.